pkg/web/initclient/init_http_handler.go

```diff
@@ -0,0 +1,61 @@
+package initclient
+
+import (
+	"errors"
+	"github.com/aerogear/mobile-security-service/pkg/web/apps"
+	"net/http"
+
+	"github.com/aerogear/mobile-security-service/pkg/httperrors"
+
+	"github.com/aerogear/mobile-security-service/pkg/models"
+	"github.com/labstack/echo"
+)
+
+type (
+	// HTTPHandler instance
+	HTTPHandler struct {
+		appsService apps.Service
+	}
+)
+
+// NewHTTPHandler returns a new instance of app.Handler
+func NewHTTPHandler(e *echo.Echo, a apps.Service) *HTTPHandler {
+	return &HTTPHandler{
+		appsService: a,
+	}
+}
+
+// InitClientApp stores device information and returns if the app version is disabled
+func (handler *HTTPHandler) InitClientApp(c echo.Context) error {
+	deviceInfo := new(models.Device)
+
+	if err := c.Bind(deviceInfo); err != nil {
+		return err
+	}
+
+	// Check the request body is valid
+	if err := validateInitRequestBody(deviceInfo); err != nil {
+		return httperrors.BadRequest(c, err.Error())
+	}
+
+	initResponse, err := handler.appsService.InitClientApp(deviceInfo)
+
+	// If no app has been found in the database, return a bad request to the client
+	if err == models.ErrNotFound {
+		return httperrors.BadRequest(c, "No bound app found for the sent App ID")
+	}
+
+	if err != nil {
+		return httperrors.GetHTTPResponseFromErr(c, err)
+	}
+
+	return c.JSON(http.StatusOK, initResponse)
+}
+
+// Request body must have version, appId and deviceId
+func validateInitRequestBody(deviceInfo *models.Device) error {
```

Comment:
It is not necessary, check in the updated versions that you can transform a JSON to a struct.

Reply:
I didn't implement this, but the purpose of it is not to transform JSON to a struct. It is checking if the device model is missing properties and returns an error if so. Come to think of it, this may be better in the model.

Reply:
It is currently specific to the init request. Not sure I would put this in the model. It is specific to the http layer in my opinion